pkgs/applications/audio/deadbeef/plugins/musical-spectrum.nix

```diff
@@ -0,0 +1,38 @@
+{ deadbeef
+, fetchFromGitHub
+, fftw
+, glib
+, gtk3
+, lib
+, pkg-config
+, stdenv
+}:
+
+stdenv.mkDerivation rec {
+  pname = "deadbeef-musical-spectrum-plugin";
+  version = "0.9";
+
+  src = fetchFromGitHub {
+    owner = "cboxdoerfer";
+    repo = "ddb_musical_spectrum";
+    rev = "a97fd4e1168509911ab43ba32d815b5489000a06";
+    sha256 = "0p33wyqi27y0q1mvjv5nn6l3vvqlg6b8yd6k2l07bax670bl0q3g";
+  };
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ deadbeef fftw glib gtk3 ];
+  makeFlags = [ "gtk3" ];
+
+  meta = with lib; {
+    description = "Musical spectrum plugin for the DeaDBeeF music player";
+    homepage = "https://github.com/cboxdoerfer/ddb_musical_spectrum";
+    license = licenses.gpl2;
+    platforms = platforms.linux;
+    maintainers = [ maintainers.ddelabru ];
+  };
+
+  installPhase = ''
```

Comment:
When overriding `installPhase`, you should also provide `runHook preInstall` and `runHook postInstall`. Though, in this case, it is better to rely on the default `installPhase`. See <https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md> for more info.

Reply:
I added calls to the `preInstall` and `postInstall` hooks. Using the default `installPhase` doesn't work for this plugin without some additional changes, because the plugin's Makefile doesn't have an install target; instead a script is supplied that would need to be modified for the purposes of this derivation.